pkg/reconciler/v1alpha1/taskrun/resources/cloudevent/cloud_event_controller_test.go

```diff
@@ -0,0 +1,251 @@
+				t.Errorf("Wrong Cloud Events Status (-want +got) = %s", diff)
+			}
+		})
+	}
+}
+
+func TestInitializeCloudEvents(t *testing.T) {
+	tests := []struct {
+		name              string
+		taskRun           *v1alpha1.TaskRun
+		pipelineResources []*v1alpha1.PipelineResource
+		wantTaskRun       *v1alpha1.TaskRun
+	}{{
+		name: "testWithMultipleMixedResources",
+		taskRun: tb.TaskRun("test-taskrun-multiple-mixed-resources", "foo",
+			tb.TaskRunSelfLink("/task/1234"), tb.TaskRunSpec(
+				tb.TaskRunTaskRef("fakeTaskName"),
+				tb.TaskRunOutputs(
+					tb.TaskRunOutputsResource("ce1", tb.TaskResourceBindingRef("ce1")),
+					tb.TaskRunOutputsResource("git", tb.TaskResourceBindingRef("git")),
+					tb.TaskRunOutputsResource("ce2", tb.TaskResourceBindingRef("ce2")),
+				),
+			),
+		),
+		pipelineResources: []*v1alpha1.PipelineResource{
+			tb.PipelineResource("ce1", "foo", tb.PipelineResourceSpec(
+				v1alpha1.PipelineResourceTypeCloudEvent,
+				tb.PipelineResourceSpecParam("TargetURI", "http://foosink"),
+			)),
+			tb.PipelineResource("ce2", "foo", tb.PipelineResourceSpec(
+				v1alpha1.PipelineResourceTypeCloudEvent,
+				tb.PipelineResourceSpecParam("TargetURI", "http://barsink"),
+			)),
+			tb.PipelineResource("git", "foo", tb.PipelineResourceSpec(
+				v1alpha1.PipelineResourceTypeGit,
+				tb.PipelineResourceSpecParam("URL", "http://git.fake"),
+				tb.PipelineResourceSpecParam("Revision", "abcd"),
+			)),
+		},
+		wantTaskRun: tb.TaskRun("test-taskrun-multiple-mixed-resources", "foo",
+			tb.TaskRunSpec(
+				tb.TaskRunTaskRef("fakeTaskName"),
+			),
+			tb.TaskRunStatus(
+				tb.TaskRunCloudEvent("http://barsink", "", 0, v1alpha1.CloudEventConditionUnknown),
+				tb.TaskRunCloudEvent("http://foosink", "", 0, v1alpha1.CloudEventConditionUnknown),
+			),
+		),
+	}, {
+		name: "testWithNoCloudEventResources",
+		taskRun: tb.TaskRun("test-taskrun-no-cloudevent-resources", "foo",
+			tb.TaskRunSelfLink("/task/1234"), tb.TaskRunSpec(
+				tb.TaskRunTaskRef("fakeTaskName"),
+				tb.TaskRunOutputs(
+					tb.TaskRunOutputsResource("git", tb.TaskResourceBindingRef("git")),
+				),
+			),
+		),
+		pipelineResources: []*v1alpha1.PipelineResource{
+			tb.PipelineResource("git", "foo", tb.PipelineResourceSpec(
+				v1alpha1.PipelineResourceTypeGit,
+				tb.PipelineResourceSpecParam("URL", "http://git.fake"),
+				tb.PipelineResourceSpecParam("Revision", "abcd"),
+			)),
+		},
+		wantTaskRun: tb.TaskRun("test-taskrun-no-cloudevent-resources", "foo",
+			tb.TaskRunSpec(
+				tb.TaskRunTaskRef("fakeTaskName"),
+			),
+			tb.TaskRunStatus(),
+		),
+	}}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			InitializeCloudEvents(tc.taskRun, tc.pipelineResources)
+			opts := GetCloudEventDeliveryCompareOptions()
+			if diff := cmp.Diff(tc.wantTaskRun.Status, tc.taskRun.Status, opts...); diff != "" {
+				t.Errorf("Wrong Cloud Events Status (-want +got) = %s", diff)
+			}
+		})
+	}
+}
+
+// TBD: TestSendCloudEvents error tests
```

Comment:
would be good to put an issue # here too, like `TODO(#111111)` (otherwise stuff like this can just kinda accumulate and itll not be clear later on what action should be taken)

Reply:
I forgot to remove this comment, it's done in `TestSendCloudEventsErrors`